### PR TITLE
added clientNote to show the sign out option

### LINKED ISSUE #20656: Reset password does not show option to sign out from other devices
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

authentication

### Describe the bug

When a user update his password using the Signing in page of the account console , an option is provided to sign out the user from existing sessions.
When the forgot option is used and the emailed link is followed to reset a password, no such option is provided.

### Version

21.1.1

### Expected behavior

The reset password link should open the Update password page with the  same option to sign out from other devices

### Actual behavior

The Update password page linked to the reset password email does not show the option to sign out from other devices

### How to Reproduce?

Log in to the console
Configure the email settings in the realm Settings>Email tab
Enable the Forgot password option in the realm settings>Login tab
Log out
Click the "Forgot password?" link and add an email address
Click on the link included in the email, which displays the Update password page

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by dmartinol**
This close #20656 
Added configuration of the client note to display the sign out option

### CODE REVIEW

File: services/src/main/java/org/keycloak/authentication/authenticators/resetcred/ResetPassword.java [hunk 1]
```diff
@@ -35,6 +38,7 @@ public void authenticate(AuthenticationFlowContext context) {
                 (context.getExecution().isConditional() &&
                         configuredFor(context))) {
             context.getAuthenticationSession().addRequiredAction(UserModel.RequiredAction.UPDATE_PASSWORD);
+            context.getAuthenticationSession().setClientNote(Constants.KC_ACTION_EXECUTING, UserModel.RequiredAction.UPDATE_PASSWORD.name());
```
Comment (pedroigor) [inline services/src/main/java/org/keycloak/authentication/authenticators/resetcred/ResetPassword.java:41]: The problem with using `Constants.KC_ACTION_EXECUTING` is that it is related to Application Initiated Actions (AIA). It is a different way of executing required actions, not really as a step of the authentication flow.

I think we need something else to indicate to both the `UpdatePassword` required action and the `login-update-password.ftl` template to handle the option to logout sessions.

In fact, can we just have it enable all the time regardless if you are running the action as part of the authentication flow or as a AIA?

Comment (dmartinol) [reply services/src/main/java/org/keycloak/authentication/authenticators/resetcred/ResetPassword.java:41]: If you mean to remove the if guard in the template I have no problem with that:
```
                        <#if isAppInitiatedAction??>
                            <div class="checkbox">
                                <label><input type="checkbox" id="logout-sessions" name="logout-sessions" value="on" checked> ${msg("logoutOtherSessions")}</label>
                            </div>
                        </#if>
```

File: services/src/main/java/org/keycloak/authentication/authenticators/resetcred/ResetPassword.java [hunk 2]
```diff
@@ -18,9 +18,12 @@
 package org.keycloak.authentication.authenticators.resetcred;
 
 import org.keycloak.authentication.AuthenticationFlowContext;
+import org.keycloak.authentication.requiredactions.UpdatePassword;
```
Comment (pedroigor) [inline services/src/main/java/org/keycloak/authentication/authenticators/resetcred/ResetPassword.java:21]: Looks like unrelated but I'm not hard on that :)

We really need to enable code style/format checks ...

**Review by pedroigor — APPROVED** [review]

**Comment by pedroigor** [comment]
@dmartinol Thanks !

**Pull request merged**